app/src/main/java/com/example/mystore/ui/login/LoginViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.mystore.ui.login
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import com.example.mystore.model.LoginModel
+import com.example.mystore.model.LoginResponse
+import com.example.mystore.repository.LoginRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import retrofit2.Response
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val repository: LoginRepository
+) : ViewModel() {
+    var response: Response<LoginResponse>? by mutableStateOf(null)
+    var loginResponse: LoginResponse? by mutableStateOf(null)
+
+    suspend fun login(loginModel: LoginModel) {
```

Comment:
If this function is fully responsible for the state management after the request has been made, it should no longer be a `suspend` one and the request should be called inside of a `viewModelScope { }`.